clippy_lints/src/manual_let_else.rs

```diff
@@ -151,10 +151,20 @@ fn emit_manual_let_else(cx: &LateContext<'_>, span: Span, expr: &Expr<'_>, pat:
             } else {
                 format!("{{ {sn_else} }}")
             };
-            let sn_bl = if matches!(pat.kind, PatKind::Or(..)) {
-                format!("({sn_pat})")
-            } else {
-                sn_pat.into_owned()
+            let sn_bl = match pat.kind {
+                PatKind::Or(..) => {
+                    let (sn_pat, _) = snippet_with_context(cx, pat.span, span.ctxt(), "", &mut app);
+                    format!("({sn_pat})")
+                },
+                PatKind::TupleStruct(ref w, ..) => {
+                    let sn_wrapper = cx.sess().source_map().span_to_snippet(w.span()).unwrap_or_default();
+                    let (sn_inner, _) = snippet_with_context(cx, local.span, span.ctxt(), "", &mut app);
+                    format!("{sn_wrapper}({sn_inner})")
+                },
```

Comment:
@giraffate @est31 Thank you for your advice:)
Finally, it's addressed by separating the `PatKind::Or` case from the `PatKind::TupleStruct` case, which we should fixed in the PR.

Could you review it again? thanks

Reply:
The code is absolutely not enough but it will improve the situation at least for the `Some` and `Ok` cases. The biggest issue I think is that tuple structs with multiple arguments are not supported, as in:

```
let foo = if let SomeEnum::Val(v, 0) = e() { v } else { _ };
```

Will be turned into:

```Rust
let SomeEnum::Val(v) = e() else { _ };
```

Reply:
But I think the PR should be accepted. :+1:

Reply:
I see the example above should be handled like below:
```rust
let SomeEnum::Val(foo, 0) = e() { v } else { ... };
               // ^ replaced ideally
```


In this PR, I [made a change](https://github.com/rust-lang/rust-clippy/pull/10175/commits/0a6f3e1c2c2d8f28f05ce1590ff5157adce58af5#diff-95ab8344127cc711fab92becc151c389f9f5bd2681c476e07475a0a6f74b35c9R275) to keep left if `TupleStruct` has more than two arguments.
I hope it's enough for now:pray: